main.go

```diff
@@ -141,13 +139,8 @@ func scrapeContainer(container types.Container, cli *client.Client, closer <-cha
 				}
 			}
 
-			restarts := inspect.RestartCount - lastRestartCount
-			if restarts < 0 {
-				restarts = 0
-			}
-			lastRestartCount = inspect.RestartCount
-			restartCounter.With(labels).Add(float64(restarts))
-
+			inspect = result.inspect
+			restartCounter.With(labels).Set(float64(inspect.RestartCount))
```

Comment:
From the API code,

```
type Counter interface {
	Metric
	Collector

	// Set is used to set the Counter to an arbitrary value. It is only used
	// if you have to transfer a value from an external counter into this
	// Prometheus metric. Do not use it for regular handling of a
	// Prometheus counter (as it can be used to break the contract of
	// monotonically increasing values).
	//
	// Deprecated: Use NewConstMetric to create a counter for an external
	// value. A Counter should never be set.
	Set(float64)
	// Inc increments the counter by 1.
	Inc()
	// Add adds the given value to the counter. It panics if the value is <
	// 0.
	Add(float64)
}
```

Reply:
Note that Set() is deprecated. :(